Ryujinx.HLE/HOS/Services/Hid/HidDevices/DebugPadDevice.cs

```diff
@@ -6,20 +9,20 @@ public class DebugPadDevice : BaseDevice
 
         public void Update()
         {
-            ref ShMemDebugPad debugPad = ref _device.Hid.SharedMemory.DebugPad;
+            ref RingLifo<DebugPadState> lifo = ref _device.Hid.SharedMemory.DebugPad;
+
+            ref DebugPadState previousEntry = ref lifo.GetCurrentEntryRef();
 
-            int currentIndex = UpdateEntriesHeader(ref debugPad.Header, out int previousIndex);
+            DebugPadState newState = new DebugPadState();
 
-            if (!Active)
+            if (Active)
             {
-                return;
+                // TODO: do something
```

Comment:
```suggestion
                // TODO: Do something
```
Could also benefit from a more detailed explanation of what needs to be done.